rllib/core/learner/learner_group.py

```diff
@@ -557,6 +551,211 @@ def _load_state(w):
+
+           This method only needs to be called if the LearnerGroup is training
+           distributed learners (e.g num_learner_workers > 0).
+
+        Args:
+            marl_module_ckpt_dir: The path to the checkpoint for the
+                MultiAgentRLModule.
+            modules_to_load: A set of module ids to load from the checkpoint.
+            rl_module_ckpt_dirs: A mapping from module ids to the path to a
+                checkpoint for a single agent RLModule.
+
+        """
+
+        assert len(self._workers) == self._worker_manager.num_healthy_actors()
+        workers = self._worker_manager.healthy_actor_ids()
+        head_node_ip = ray.util.get_node_ip_address()
+
+        def _load_module_state(w):
+            # doing imports here since they might not be imported on the worker
+            import ray
+            import tempfile
+            import shutil
+
+            worker_node_ip = ray.util.get_node_ip_address()
+            # sync the checkpoints from the head to the worker if the worker is not
+            # on the same node as the head
+            tmp_marl_module_ckpt_dir = marl_module_ckpt_dir
+            tmp_rl_module_ckpt_dirs = rl_module_ckpt_dirs
+            if worker_node_ip != head_node_ip:
+                if marl_module_ckpt_dir:
+                    tmp_marl_module_ckpt_dir = tempfile.mkdtemp()
+                    sync_dir_between_nodes(
+                        source_ip=head_node_ip,
+                        source_path=marl_module_ckpt_dir,
+                        target_ip=worker_node_ip,
+                        target_path=tmp_marl_module_ckpt_dir,
+                    )
+                if rl_module_ckpt_dirs:
+                    tmp_rl_module_ckpt_dirs = {}
+                    for module_id, path in rl_module_ckpt_dirs.items():
+                        tmp_rl_module_ckpt_dirs[module_id] = tempfile.mkdtemp()
+                        sync_dir_between_nodes(
+                            source_ip=head_node_ip,
+                            source_path=path,
+                            target_ip=worker_node_ip,
+                            target_path=tmp_rl_module_ckpt_dirs[module_id],
+                        )
+                        tmp_rl_module_ckpt_dirs[module_id] = pathlib.Path(
+                            tmp_rl_module_ckpt_dirs[module_id]
+                        )
+            if marl_module_ckpt_dir:
+                # load the MARLModule checkpoint if they were specified
+                w.module.load_state(
+                    marl_module_ckpt_dir, modules_to_load=modules_to_load
+                )
+            if rl_module_ckpt_dirs:
+                # load the RLModule if they were specified
+                for module_id, path in rl_module_ckpt_dirs.items():
+                    w.module[module_id].load_state(path / RLMODULE_STATE_DIR_NAME)
+
+            # remove the temporary directories on the worker if any were created
```

Comment:
curious, do you really need to remove these, given that we used tempfile for them?

Reply:
they are a tempfile, but doesn't /tmp/ only get cleared after one week?

I'm not using the with: scope for the tempfile, so the directories won't be automatically removed by tempfile library.